### ISSUE TITLE
Empty Roslyn project in net standard library.

### ISSUE THREAD

**Issue by salmelo** [daveaglick/Scripty#91]
Using the MSBuild task in a .NET Standard class library, `Project.Analysis.Documents` is empty, as far as I can tell there's nothing in the Roslyn project at all.

Dunno if this is anything on Scripty's end or if anything can be done about it, but figured it was worth making known.

**Comment by daveaglick** [comment]
If there's stuff in the project, it should be in the Roslyn workspace, and Roslyn shouldn't have any issues dealing with .NET Standard/Core projects. In other words, there should be documents in the collection.

Can you provide any more information? What does the project structure look like?

**Comment by daveaglick** [comment]
Any updates on this? Did you get it figured out?

**Comment by kentcb** [comment]
Hi,

I came here expecting this exact kind of bug report. My understanding is that Roslyn is _not_ capable of working with SDK-style csproj. See [here](https://github.com/dotnet/roslyn/issues/17968). It has blocked me from updating my own software libraries to work correctly in the new world.

**Comment by josephwoodward** [comment]
Yeah, currently a blocker for me on my [IL Viewer for VS Code](https://github.com/JosephWoodward/VSCodeILViewer/) too :(

**Comment by daveaglick** [comment]
Hmm...I wonder if the custom workspace the Mono folks built for VS Mac would work

**Comment by josephwoodward** [comment]
What custom workspace did they build? I'd be interested in trying it. Would love to get my IL Viewer working again.

**Comment by daveaglick** [comment]
@JosephWoodward I'm going off of this comment: https://github.com/dotnet/roslyn/issues/20275#issuecomment-309100824 Apparently OmniSharp also has it's own workspace (a little further down). Been meaning to check either/both of those out.

Scripty has other issues with MSBuild workspace like that it can't be used as an MSBuild task because MSBuild won't let you nest execution. Would love to get the workspace (and thus Scripty itself) in-proc with the build task.

**Comment by josephwoodward** [comment]
@daveaglick That's right, OmniSharp have their own adhoc workspace that they've built from the ground up which resolves all dependencies. It's pretty comprehensive. I've considered trying to build one but having looked at it I wouldn't know where to start.
